File: REACT/testapp/src/Catalog/Components/Gallery/Product.js
import React from 'react';
import Counter from './Counter';

const Product = ({ product, onLike,onUnlike }) => {

  return (
    <div>
      <img src={product.imageurl} width="100" height="100"/>
      <h2>{product.title}</h2>
      <p>Price: ${product.unitprice}</p>
      <p>likes: {product.likes}</p>
      <Counter likes={product.likes} onCounterLike={() => onLike(product.id)}
                                     onCounterUnLike={() => onUnlike(product.id)}></Counter>
    </div>
  );
};

export default Product;

File: REACT/React-Application/Back-End/Entities/Product.cs
 
namespace ProductApi.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public decimal UnitPrice{get;set;}
        public int Quantity {get; set;}
        public int Likes {get;set;}
        public string ImageUrl{get;set;}
    }
}
